### ISSUE TITLE
5.0.2 Memory leak on OSX?

### ISSUE THREAD

**Issue by variable** [mattermost/desktop#1998]
I confirm (by marking "x" in the [ ] below: [x]):

- [x] This is not a troubleshooting question. [Troubleshooting questions go here: http://www.mattermost.org/troubleshoot/](http://www.mattermost.org/troubleshoot/).
- [ ] This doesn't reproduce on web browsers (such as in Chrome). If it does, [issue reports go to the Mattermost Server repository](https://github.com/mattermost/platform/issues).
- [x] I have read [contributing guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md).

---

**Summary**
Running MM for a few days, with opening different channels etc result excessive memory consumption

**Environment**
- Operating System: Mac OSX 11.6
- Mattermost Desktop App version: 5.0.2
- Mattermost Server version: X.X.X 

**Steps to reproduce**
Run MM for a few days and browse different channels

**Expected behavior**
Runs smooth and doesn't consume a lot of memory

**Observed behavior**
Excessive memory usage

![image](https://user-images.githubusercontent.com/558175/152872970-a38f8a2d-768b-46d0-b54d-8a1dfc415506.png)

**Comment by devinbinnie** [comment]
@variable Do you experience similar behaviour running in Chrome? I'm trying to establish whether this is something the Desktop App is doing or if it's the web app.

**Comment by sadohert** [comment]
@devinbinnie * Recent upgrade from Mattermost Desktop `5.0.3` to `5.0.4`.
* MacBook Pro 16-inch, 16 GB RAM, 
* MacOS Monterey/12.1
* See Mattermost Memory usage here:
![image](https://user-images.githubusercontent.com/260544/153408174-f7694c73-92ba-4142-b94d-a176e949f31f.png)
* See Mattermost CPU:
![image](https://user-images.githubusercontent.com/260544/153408324-92bf6cba-693f-4c0f-ac93-45dc494eb53b.png)
* Configured to point to `Community Daily` Mattermost environment.

Immediately after the upgrade to `5.0.4` the app seemed to perform well.  

Now, after 24-48 hours, I'm seeing slow typing response and long channel switching times.

When trying to compare similar interactions with the browser tab (as of this moment) I see channel switching is somewhat slow, but I think the effect on the desktop app is much more pronounced.

Anything else I can provide?

**Comment by Willyfrog** [comment]
> Immediately after the upgrade to 5.0.4 the app seemed to perform well.

I've noticed this as well, I was waiting a bit more to confirm, but since I upgraded (2 days ago at most) I haven't seen any locks.

**Comment by devinbinnie** [comment]
> > Immediately after the upgrade to 5.0.4 the app seemed to perform well.
> 
> I've noticed this as well, I was waiting a bit more to confirm, but since I upgraded (2 days ago at most) I haven't seen any locks.

My tests were performed on `master` using the latest version of Electron (v17). I'd be surprised if it didn't happen on v5.0.4.

**Comment by sadohert** [comment]
@devinbinnie 

> So I left my MacBook running all night with Mattermost open in both Google Chrome and the desktop app, and the memory usage is about the same for both of them.

Is that usage growing for you?

**Comment by devinbinnie** [comment]
@sadohert Yes, Electron is now at 1506M and Chrome is now at 1412M. Both were sitting around 700ish when I signed off yesterday.

**Comment by variable** [comment]
> @variable Do you experience similar behaviour running in Chrome? I'm trying to establish whether this is something the Desktop App is doing or if it's the web app.
> 
> Also, is there any more to your screenshot? More specifically, is the process `Mattermost Helper (Renderer)`?

Hi sorry for the late reply, I haven't managed to compared with browser, but I have had this issue for a long time even before 5.0.2, saying that my chrome is also having memory issue tho at a much longer timespan.

**Comment by devinbinnie** [comment]
Thanks @variable.

At this point, I think it's safe to say that this is a webapp issue, so I'm going to transfer the issue over.
